src/overlays/actors/ovl_En_Ending_Hero/z_en_ending_hero.c

```diff
@@ -21,16 +23,73 @@ const ActorInit En_Ending_Hero_InitVars = {
     (ActorFunc)EnEndingHero_Update,
     (ActorFunc)EnEndingHero_Draw
 };
-*/
 
-GLOBAL_ASM("asm/non_matchings/ovl_En_Ending_Hero_0x80C1E690/EnEndingHero_Init.asm")
+extern SkeletonHeader D_0600B0CC;
+extern AnimationHeader D_06000BE0;
+
+void EnEndingHero_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnEndingHero* this = THIS;
+
+    this->actor.unkA0.mass = 0xFF;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.unk1F = 6;
+    this->actor.gravity = -3.0f;
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600B0CC, &D_06000BE0, this->limbDrawTable,
+                     this->transitionDrawTable, 15);
+    Actor_SetDrawParams(&this->actor.shape, 0.0f, func_800B3FC0, 25.0f);
+    func_80C1E748(this);
+}
+
+void EnEndingHero_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void func_80C1E748(EnEndingHero* this) {
+    this->unk244 = 1;
+    this->actionFunc = func_80C1E764;
+}
+
+void func_80C1E764(EnEndingHero* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+void EnEndingHero_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnEndingHero* this = THIS;
+
+    if (this->unk240 == 0) {
+        this->unk242 += 1;
+        if (this->unk242 > 2) {
+            this->unk242 = 0;
+            this->unk240 = (s16)randZeroOneScaled(60.0f) + 0x14;
+        }
+    }
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    func_800B78B8(globalCtx, &this->actor, 20.0f, 20.0f, 50.0f, 0x1D);
+}
+
+UNK_PTR D_80C1E970[] = { 0x06007350, 0x06009590, 0x06009F90, 0x0600A790, 0x0600AB90 };
```

Comment:
These look to be textures that come from the object file.
We don't currently have types for textures, but you should at least make a symbol for each of these segmented pointers if it's still OK.